### DISCUSSION TITLE
WSA 2307 Update!

### DISCUSSION THREAD

**Discussion by hamza-usmani** [microsoft/WSA#374] [Announcements]
# What's New

We've shipped an update for Windows Subsystem for Android™ on Windows 11 to the [Windows Subsystem for Android™ Preview Program](https://learn.microsoft.com/en-us/windows/android/wsa/preview-program). This update (2307.40000.2.0) includes improvements to advanced settings, graphics reliability and several requested features.

## Release Notes

- Custom Microsoft.Windows.LAUNCH_URI intent to launch apps on Windows via protocol activation
- Platform stability improvements
- We’ve added the ability for apps to use the Vulkan API in preview, options are available in the settings app to try this out
- Graphics reliability improvements
- Updated latest Chromium WebView to version 115
- Android 13 security updates

# Giving feedback

If you are having issues with Windows Subsystem for Android™ – please file feedback via Feedback Hub under Apps > Windows Subsystem for Android™. For more information about troubleshooting and submitting feedback: [Troubleshooting and FAQ for mobile apps on Windows](https://support.microsoft.com/en-us/windows/troubleshooting-and-faq-for-mobile-apps-on-windows-d6062afd-98a1-4018-a7c8-6b3b680a2ea5).

Thanks,
Windows Subsystem for Android™ Team

**Comment by HotCakeX** [comment]
I thought with this update adding Vulkan API, Edge browser would work in WSA, but still the same. Edge has never worked in WSA..

<br>

![image](https://github.com/microsoft/WSA/assets/118815227/9f17421e-4924-464b-a3a6-45843cd5aa3d)

**Reply by zavocc** [reply]
Kiwi Browser and Firefox are the working browsers in WSA as I known to use it

**Reply by akhil-rana** [reply]
Brave also works. It's also chromium based. Not sure why edge doesn't work.

**Reply by nieubank** [reply]
Can you file a feedback hub report for this or share a logcat under bugs? I just tried Edge and it works fine in my setup.

**Reply by joshieecs** [reply]
Edge for Android only ships packages for ARM targets, but none for x86. I don't know why. Chromium does some things that are not compatible with the ARM translation WSA uses (Intel Bridge Technology). Other browsers are probably working because they ship packages for x86 targets. You can see issue #249 marked by-design.

**Reply by HotCakeX** [reply]
So it's by design,
The developer's comment makes sense,
I actually never browse the web from WSA because the UI, features, performance etc. in Edge desktop is much better than any WSA browser.

iirc WSA has Chromium WebView and apps that use that instead of their own engine work fine.

**Comment by HotCakeX** [comment]
> Can you file a feedback hub report for this or share a logcat under bugs? I just tried Edge and it works fine in my setup.

This is the feedback i just filed
https://aka.ms/AAlwvn0

**Comment by bubbleguuum** [comment]
With this version, my app BubbleUPnP available on the Amazon appstore whose purpose is to play to networked streamers (Chromecast, UPnP/DLNA, ...) is unable to detect these streamers using multicast SSDP discovery (and mDNS).

In previous versions, multicast SSDP discovery worked fine after enabling the "Advanced networking" setting, which is gone.
It looks like it may have been replaced with "Advanced settings > Experimental features > Local network access", but enabling it does not make SSDP discovery work.

Is it a known issue ? Not working SSDP discovery renders my app almost useless.

**Reply by ColinWillies** [reply]
I have been tearing my hair out trying to get Chromecast to work for a different app from WSA - now I think I'll wait for the next update or a reaction to this query. Its a shame to have to use my phone as a Chromecast source when I have a perfectly good PC running WSA that could do it much more easily, leaving my phone free for other things (like phonecalls)...

**Comment by HiZhixiang** [comment]
Hey,I'm glad to see the update for WSA. We hope to support Key/Mouse mapping, Root Access , and Google Services as soon as possible, so that we can surpass other Android emulators in terms of basic experience.💪

**Comment by omidsky2600** [comment]
Support keyboard windows devices with touchscreen ?
Don't Support keyboard touchscreen

**Comment by omidsky2600** [comment]
why doesn't it show the keyboard when l installed the keyboard app ?
I installed. The. Google keyboard. But it does. Not. Show the letters
Every keyboard program I install is the same
There was no such problem before when there was Android 12
This happened when Android 13 was released

**Comment by Gerdya** [comment]
None of the Vulkan options seem to work on ARM64 devices. What could be the issue?

**Reply by nieubank** [reply]
Most ARM64 device's have drivers that are not fully compatible with D3D12, thus the D3D12 option causes a crash at boot. ARM64 devices also don't have native drivers so that option does not work either. We're working on making this clearer in future versions, Vulkan is still in the experimental testing phase.

**Reply by Gerdya** [reply]
Thanks. Could you please give a few more details? On one of my machines, where I have an ICD installed, WSA starts but apps are crashing inside the ICD. On another machine without ICD, WSA crashes at startup - as you said. 
Also in the first case, where the ICD is crashing it does not seem related to D3D12 - it is crashing in a hash function, where an invalid key is provided.

**Reply by nieubank** [reply]
Filing a feedback hub is the best way we can help at this time. We are tracking the reliability issues around Vulkan start up with internal telemetry so this is something that will be resolved with time as the feature is used more. This is primarily why we have the feature disabled and opt-in at the moment.

**Reply by Gerdya** [reply]
Thanks for feedback! Will file an issue via feedback hub as you suggested.  I also reported the issue to  https://gitlab.freedesktop.org/mesa/mesa/-/issues/9600
